examples/platforms/efr32/uart.c

```diff
@@ -146,16 +196,26 @@ otError otPlatUartEnable(void)
 {
     UARTDRV_Init_t uartInit = USART_INIT;
 
-    sReceiveFifo.mHead = 0;
-    sReceiveFifo.mTail = 0;
+    memset(&sRxNvicMask, 0, sizeof(sRxNvicMask));
+    CORE_NvicMaskSetIRQ(LDMA_IRQn, &sRxNvicMask);
+    CORE_NvicMaskSetIRQ(USART_PORT_RX_IRQn, &sRxNvicMask);
+
+    sReceiveFifo.mReadStart = 0;
+    sReceiveFifo.mReadEnd   = 0;
+    sReceiveFifo.mWrite     = 0;
 
     UARTDRV_Init(sUartHandle, &uartInit);
 
-    for (uint8_t i = 0; i < sizeof(sReceiveBuffer); i++)
+    CORE_DECLARE_NVIC_STATE;
+    CORE_ENTER_NVIC(&sRxNvicMask);
+
+    for (int i = 0; i < 2; i++)
```

Comment:
I assume the original 2 byte sReceiveBuffer was there to flush the 2 byte Rx buffer inside the UART when calling otPlatUartEnable.

Will calling queueNextReceive() discard the data or place it in the sReceiveFifo buffer as read data?

Reply:
Thanks for reviewing and testing!
I think the original `sReceiveBuffer` was the actual receive buffer used by the USART. In the `receiveDone` interrupt handler, the single byte was then manually copied in to the ringbuffer `sReceiveFifo`. It was initially called twice in this initialisation function to set up double buffering. 

Is there any reason we should in fact drop the first 2 bytes in the USART at initialisation?

Reply:
There's no particular reason to clear from what I understand.  I just had a feeling it may have been to flush whatever was currently there in the queue at startup but don't think it will be an issue.